src/Compilers/CSharp/Portable/Symbols/Source/ThisParameterSymbol.cs

```diff
@@ -40,10 +40,22 @@ public override RefKind RefKind
         {
             get
             {
-                return
-                    ((object)_containingType == null || _containingType.TypeKind != TypeKind.Struct) ? RefKind.None :
-                    ((object)_containingMethod != null && _containingMethod.MethodKind == MethodKind.Constructor) ? RefKind.Out :
-                    RefKind.Ref;
+                if (ContainingType?.TypeKind != TypeKind.Struct)
+                {
+                    return RefKind.None;
+                }
+
+                if (_containingMethod?.MethodKind == MethodKind.Constructor)
```

Comment:
Still thinking if this can be tested in some reasonable scenario, other than just asking someMethod.ThisParameter.RefKind  directly. I guess, I'll add just that.

---
In reply to: [118578857](https://github.com/dotnet/roslyn/pull/19713#discussion_r118578857) [](ancestors = 118578857,118576819,118565720,118552148)